backend/utils/graph_db_utils_test.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import ANY, MagicMock
+from neo4j import Driver, Session
+import pytest
+import utils.graph_db_utils
+
+def test_database_connectivity_is_healthy(mocker):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_driver.verify_connectivity.return_value = None
+
+    connected = utils.graph_db_utils.test_connection()
+
+    assert connected
+    mock_driver.verify_connectivity.assert_called_once()
+    mock_driver.close.assert_called_once()
+
+
+def test_database_connectivity_is_unhealthy(mocker):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_driver.verify_connectivity.side_effect = Exception
+
+    connected = utils.graph_db_utils.test_connection()
+
+    assert not connected
+    mock_driver.verify_connectivity.assert_called_once()
+    mock_driver.close.assert_called_once()
+
+
+def test_create_goal_is_successful(mocker):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_session = MagicMock(spec=Session)
+    mock_driver.session.return_value = mock_session
+
+    response = utils.graph_db_utils.create_goal("Test Name", "Test Description")
+
+    assert response is None
+    mock_session.run.assert_called_once_with(ANY, name="Test Name", description="Test Description")
```

Comment:
Should we use something more narrow than `ANY` ? Can we assert against a value?

Reply:
It's tricky to do this within `assert_called_once_with` as we'd need to exactly match the generated cypher query string, which has a combination of new lines and whitespace. I found it hard to exactly replicate this string in a test, which is why I used `ANY`.

Instead, I've used `call_args` to extract the arguments passed to the method. I can then remove the random whitespace and newlines and do the assertions in separate statements.